### PR TITLE
Implement Pagination and Sorting

### PR DESCRIPTION
**Pull request by gajendraJangid421**
Added pagination, sorting and token expiry

### CODE REVIEW

File: event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 1]
```diff
@@ -19,6 +20,17 @@ public List<Events> findAll() {
         return eventsService.findAll();
     }
 
+    //for admin and user
+    @GetMapping(path = "/page")
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/controller/EventsController.java]:
```suggestion
    @GetMapping(path = "/search")
```

File: event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 2]
```diff
@@ -21,14 +23,24 @@ public List<Users> findAll() {
         return usersService.findAll();
     }
 
+    //for admin
+    @GetMapping(path = "/page")
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/controller/UsersController.java]:
```suggestion
    @GetMapping(path = "/search")
```

File: event_management/src/main/java/com/example/event_management/exception/UnAuthorisedException.java [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
+import lombok.Getter;
+import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import static org.springframework.http.HttpStatus.UNAUTHORIZED;
 
+@Getter
 @ResponseStatus(code = UNAUTHORIZED)
 public class UnAuthorisedException extends RuntimeException {
+
+    private HttpStatus httpStatus= UNAUTHORIZED;
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/exception/UnAuthorisedException.java]:
```suggestion
    private HttpStatus httpStatus = UNAUTHORIZED;
```

File: event_management/src/main/java/com/example/event_management/repository/SessionRepository.java [hunk 4]
```diff
@@ -2,8 +2,15 @@
...
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.util.Optional;
 
 public interface SessionRepository extends JpaRepository<Session, String> {
 
     void deleteByToken(String token);
+
+    @Query("SELECT s FROM Session s WHERE s.token =:requestToken")
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/repository/SessionRepository.java]:
```suggestion
    @Query("SELECT s FROM Session s WHERE s.token = :requestToken")
```

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    UsersRepository usersRepository;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws IOException, ServletException {
+
+        String requestToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+        String path=request.getRequestURI();
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]:
```suggestion
        String path = request.getRequestURI();
```

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+        if(path.equals("/api/login") || path.equals("/api/users")){
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]: Only login API should work without token
```suggestion
        if(path.equals("/api/login")){
```

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+            if (Objects.nonNull(requestToken) && !requestToken.isEmpty()) {
+                UsernamePasswordAuthenticationToken authentication = getAuthentication(request,response);
+                SecurityContextHolder.getContext().setAuthentication(authentication);
+            }
+
+            filterChain.doFilter(request, response);
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]: No need to validate token during login request
```suggestion
            filterChain.doFilter(request, response);
```

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+            try {
+
+                if (requestToken != null && !requestToken.isEmpty()) {
+                    UsernamePasswordAuthenticationToken authentication = getAuthentication(request,response);
+                    SecurityContextHolder.getContext().setAuthentication(authentication);
+                    filterChain.doFilter(request, response);
+                }
+                else {
+                    throw new UnAuthorisedException("Token not found");
+                }
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]:
```suggestion
                if (StringUtils.isBlank(requestToken)) {
                    throw new UnAuthorisedException("Auth token is required");
                } else {
                    UsernamePasswordAuthenticationToken authentication = getAuthentication(request,response);
                    SecurityContextHolder.getContext().setAuthentication(authentication);
                    filterChain.doFilter(request, response);
                }
```

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+            }
+            catch (UnAuthorisedException e){
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]:
```suggestion
            } catch (UnAuthorisedException e){
```

File: event_management/src/main/java/com/example/event_management/service/EventsService.java [hunk 10]
```diff
@@ -19,11 +23,18 @@ public class EventsService {
     @Autowired
     private UserEventService userEventService;
 
-
     public List<Events>  findAll() {
         return eventsRepository.findAll();
     }
 
+    public List<Events>  findEventsWithPagination(int offset, int limit, String sortBy) {
+        Sort sort = Sort.by(Sort.Direction.ASC, sortBy);
```
Comment (sagarzond) [inline event_management/src/main/java/com/example/event_management/service/EventsService.java:31]: take sorting directions from user request.

File: event_management/src/main/java/com/example/event_management/controller/EventsController.java (hunk 1 above)
Comment (gajendraJangid421) [reply event_management/src/main/java/com/example/event_management/controller/EventsController.java]: updated

File: event_management/src/main/java/com/example/event_management/controller/UsersController.java (hunk 2 above)
Comment (gajendraJangid421) [reply event_management/src/main/java/com/example/event_management/controller/UsersController.java]: updated

File: event_management/src/main/java/com/example/event_management/exception/UnAuthorisedException.java (hunk 3 above)
Comment (gajendraJangid421) [reply event_management/src/main/java/com/example/event_management/exception/UnAuthorisedException.java]: updated

File: event_management/src/main/java/com/example/event_management/repository/SessionRepository.java (hunk 4 above)
Comment (gajendraJangid421) [reply event_management/src/main/java/com/example/event_management/repository/SessionRepository.java]: updated

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java (hunk 5 above)
Comment (gajendraJangid421) [reply event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]: updated

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java (hunk 8 above)
Comment (gajendraJangid421) [reply event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]: updated

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java (hunk 9 above)
Comment (gajendraJangid421) [reply event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]: updated

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java (hunk 7 above)
Comment (gajendraJangid421) [reply event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]: updated

File: event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java (hunk 6 above)
Comment (gajendraJangid421) [reply event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java]: changed /api/users to /api/users/sign-up and also added /api/users/forget-password

**Review by sagarzond — APPROVED** [review]

**Pull request merged**